boa/src/builtins/console.rs

```diff
@@ -32,11 +32,36 @@ pub fn error(_: &Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
     Ok(Gc::new(ValueData::Undefined))
 }
 
+/// Print a javascript value to the standard error if first argument evaluates to false or there were no arguments
+/// 
+/// https://console.spec.whatwg.org/#assert
+pub fn assert(_: &Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
+    let assertion = if !args.is_empty() {
+        from_value::<bool>(args[0].clone()).expect("Could not convert to bool.")
+    } else {
+        false
+    };
```

Comment:
We can improve this even further:
```suggestion
    let assertion = args.get(0).cloned().map(|val| from_value::<bool>(val).expect("Could not convert to bool.")).unwrap_or_default();
```

This of course needs a `cargo fmt`. Still, I find it a bit legacy that we use the `from_value()` function, instead of the `TryFrom` and `From` traits from the standard library. I will try to do a PR in the future to improve this. Ideally, we should also allow doing this without cloning, to avoid some allocations.

Reply:
Looks a bit cryptic to me, but surely it's because of my lack of knowledge. I will read about it.

Reply:
> Looks a bit cryptic to me, but surely it's because of my lack of knowledge. I will read about it.

What it does is instead of checking if we have an argument value first, it just tries to get it. Rust already checks bounds on slice access, and returns an `Option`.

We need the cloned value in case it has any, so that's why we call `cloned()`.

Then, we need to convert it to a `bool`, so we call `map()` with a closure that does the conversion.

Finally, since up to now we have only modified the value if it was `Option::Some(_)`, we must tell it what to do in the case it was `None`. So we call `unwrap_or_default()`. This will return the `bool` value of the `map()` function if it's `Some`, and use the `bool::default()` if it's `None`. The default value for a boolean in Rust is `false`.

Reply:
Thanks for explanation! So out-of-bound access to slice with `get` will return `None` that will be then passed all the way to `unwrap_or_default`, which will return `false`. Clever!